src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -132,8 +148,15 @@ String generateDeclarations(List<SourceFile> sourceFiles, List<SourceFile> exter
     Compiler compiler = new Compiler();
     compiler.disableThreads();
     CompilerOptions compilerOptions = opts.getCompilerOptions();
+    // Capture all errors directly instead of relying on JSCompiler's error reporting mechanisms.
+    compilerOptions.setErrorHandler(new ErrorHandler() {
+      @Override
+      public void report(CheckLevel level, JSError error) {
+        errors.add(error);
+      }
+    });
     compiler.setPassConfig(new DefaultPassConfig(compilerOptions));
-    // Don't print anything, throw later below.
+    // In spite of setting an ErrorHandler above, this is still needed to silence the console.
```

Comment:
Well, obviously the `ErrorHandler` handles the errors, and then the `ErrorManager` manages the result of the handler. Let me draw you an orgchart of `Error Shenanigans Inc.` ;-)